synapse/server.py

```diff
@@ -423,31 +420,6 @@ def build_http_client(self):
         )
         return MatrixFederationHttpClient(self, tls_client_options_factory)
 
-    def build_db_pool(self):
```

Comment:
I'm not convinced you've found all the refs to this. My search finds 7 existing occurrences to `get_db_pool` (including tests), you've only updated one so far.